src/Modules/UngradedModules/FreeMod.jl

```diff
@@ -75,6 +75,7 @@ free_module(R::MPolyRing, p::Int, name::VarName = :e; cached::Bool = false) = Fr
 free_module(R::MPolyQuoRing, p::Int, name::VarName = :e; cached::Bool = false) = FreeMod(R, p, name, cached = cached)
 free_module(R::MPolyLocRing, p::Int, name::VarName = :e; cached::Bool = false) = FreeMod(R, p, name, cached = cached)
 free_module(R::MPolyQuoLocRing, p::Int, name::VarName = :e; cached::Bool = false) = FreeMod(R, p, name, cached = cached)
+free_module(R::NCRing, p::Int, name::VarName = :e; cached::Bool = false) = FreeMod(R, p, name, cached = cached)
```

Comment:
If we do this then we can effectively delete the four methods above as they are all special cases no of this one. And also adjust the docstring.

But then ideally we also test this over a bunch of other rings `R`, e.g. `ZZ`, `QQ`, `cylotomoic_field(10)`, a free associative algebra, etc.

But I wonder how complete and correct it will be in all those cases, e.g. when `R = ZZ` ?

Reply:
As was pointed out elsewhere, for now this is really intended for PBW algebras, so I'd restrict the type here.

Also, looking at the code further, one very quickly runs into code that insists on using `ModuleGens`, which means this free module implementation in its current form is quite closely tied to rings backed by Singular, which is another reason to restrict it here for now.

Reply:
Sure `ModuleGens` has features for Singular double book-keeping, but you don't have to to use it (e.g. over localizations we don't). In the end, one has to say how to do the curcial toolbox (kernel/lift etc) by implementing the new signatures. Then everything works (surely it can also work over `ZZ` using Smith, or over fields, indeed we use it over fields for cohomology). 

However, in the setting we have in mind here the implementation is very similar to the commutative polynomial case and will be driven by the Buchberger machinery of Singular. So one can even do the implementation one abstraction layer lower, implementing the Buchberger toolbox (normal form, std, liftstd, syz etc). Certainly some issues will show up by going to the non-commutative setting, but nothing serious, everything is very similar to the commutative setting.

Reply:
I agree that we should add to the signatures, instead of generalizing to `NCRing`.